src/ip/mod.rs

```diff
@@ -1,3 +1,11 @@
+#[derive(Debug, PartialEq)]
+struct NetworkAddr<T>(T);

```

Comment:
I've tried to understand what this is doing.  I understand that derive is defining traits for the variable (type-partials), and you are telling the compiler that your variables defined with `NetworkAddr` have these traits, but why?  Are you using these traits somewhere?  This being my first Rust program I don't understand what the purpose of this syntax.  Can you elaborate?